staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -1105,6 +1105,9 @@ type ManagedFieldsEntry struct {
 	// Time is timestamp of when these fields were set. It should always be empty if Operation is 'Apply'
 	// +optional
 	Time *Time `json:"time,omitempty" protobuf:"bytes,4,opt,name=time"`
+	// FieldsType identifies the format of raw bytes in Fields below.
+	// There is currently only one possible value: "json+v1"
+	FieldsType string `json:"fieldsType,omitempty" protobuf:"bytes,6,name=fieldsType`
```

Comment:
Unions/one-of are a thing in our apis (this was my suggested versioning mechanism). RawExtensions where the contents are self-describing (e.g., via apiVersion+kind fields in JSON or via envelope in proto) are a thing (this was @deads2k's suggested mechanism).

However, having one field explain the type of another field is not a thing as far as I know? Can you point to an example of this? I think both of the other patterns are better.